### ISSUE TITLE
Toml `ToModel` error when there is a reference in toml

### ISSUE THREAD

**Issue by FrankYang0529** [deislabs/hippo#978]
Hippo version: `v0.17.0`
Spin version: `canary` on commit `ceeccf9`

Run the test case in [spin/tests/http/simple-spin-rust/spin.toml](https://github.com/fermyon/spin/blob/ceeccf937c060fc44a531f6adc35842a37ed2ebd/tests/http/simple-spin-rust/spin.toml#L8-L9).

```console
cd spin
cargo build
cd tests/http/simple-spin-rust
../../../target/debug/spin deploy --file spin.toml --bindle-server=http://localhost:8080/v1 --hippo-server=http://localhost:5309 --hippo-username=admin --hippo-password=password --deploy-existing-bindle
```

Hippo error log:
```
fail: Hippo.Application.Apps.Commands.CreateAppCommand[0]
      Hippo Request: Unhandled Exception for Request CreateAppCommand Hippo.Application.Apps.Commands.CreateAppCommand
      Tomlyn.TomlException: (4,2) : error : Unable to set the property config on object type Hippo.Application.Common.Interfaces.BindleService.RevisionSpinToml.

         at Tomlyn.Toml.ToModel[T](DocumentSyntax syntax, TomlModelOptions options)
         at Hippo.Infrastructure.Services.BindleService.ParseSpinTomlParcel(HttpContent parcelResponse) in /Users/frankyang/Project/hippo/src/Infrastructure/Services/BindleService.cs:line 67
         at Hippo.Infrastructure.Services.BindleService.GetSpinTomlParcel(Invoice invoice, String revisionId) in /Users/frankyang/Project/hippo/src/Infrastructure/Services/BindleService.cs:line 50
         at Hippo.Infrastructure.Services.BindleService.GetRevisionDetails(String revisionId) in /Users/frankyang/Project/hippo/src/Infrastructure/Services/BindleService.cs:line 27
         at Hippo.Application.Revisions.EventHandlers.UpdateRevisionDetails.Handle(CreatedEvent`1 notification, CancellationToken cancellationToken) in /Users/frankyang/Project/hippo/src/Application/Revisions/EventHandlers/UpdateRevisionDetails.cs:line 33
         at MediatR.Mediator.PublishCore(IEnumerable`1 allHandlers, INotification notification, CancellationToken cancellationToken)
         at MediatR.MediatorExtensions.DispatchDomainEvents(IMediator mediator, DbContext context) in /Users/frankyang/Project/hippo/src/Infrastructure/Common/MediatorExtensions.cs:line 22
         at Hippo.Infrastructure.Data.ApplicationDbContext.SaveChangesAsync(CancellationToken cancellationToken) in /Users/frankyang/Project/hippo/src/Infrastructure/Data/ApplicationDbContext.cs:line 60
         at Hippo.Application.Revisions.Commands.ImportRevisionsCommandHandler.Handle(ImportRevisionsCommand request, CancellationToken cancellationToken) in /Users/frankyang/Project/hippo/src/Application/Revisions/Commands/ImportRevisionsCommand.cs:line 49
         at Hippo.Application.Common.Behaviours.PerformanceBehaviour`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next) in /Users/frankyang/Project/hippo/src/Application/Common/Behaviours/PerformanceBehaviour.cs:line 31
         at Hippo.Application.Common.Behaviours.ValidationBehaviour`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next) in /Users/frankyang/Project/hippo/src/Application/Common/Behaviours/ValidationBehaviour.cs:line 35
         at Hippo.Application.Common.Behaviours.AuthorizationBehaviour`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next) in /Users/frankyang/Project/hippo/src/Application/Common/Behaviours/AuthorizationBehaviour.cs:line 78
         at Hippo.Application.Common.Behaviours.UnhandledExceptionBehaviour`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next) in /Users/frankyang/Project/hippo/src/Application/Common/Behaviours/UnhandledExceptionBehaviour.cs:line 19
         at MediatR.Pipeline.RequestExceptionProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
         at MediatR.Pipeline.RequestExceptionProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
         at MediatR.Pipeline.RequestExceptionActionProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
         at MediatR.Pipeline.RequestExceptionActionProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
         at MediatR.Pipeline.RequestPostProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
         at MediatR.Pipeline.RequestPreProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
         at Hippo.Application.Apps.EventHandlers.InitialRevisionImport.Handle(CreatedEvent`1 notification, CancellationToken cancellationToken) in /Users/frankyang/Project/hippo/src/Application/Apps/EventHandlers/InitialRevisionImport.cs:line 26
         at MediatR.Mediator.PublishCore(IEnumerable`1 allHandlers, INotification notification, CancellationToken cancellationToken)
         at MediatR.MediatorExtensions.DispatchDomainEvents(IMediator mediator, DbContext context) in /Users/frankyang/Project/hippo/src/Infrastructure/Common/MediatorExtensions.cs:line 22
         at Hippo.Infrastructure.Data.ApplicationDbContext.SaveChangesAsync(CancellationToken cancellationToken) in /Users/frankyang/Project/hippo/src/Infrastructure/Data/ApplicationDbContext.cs:line 60
         at Hippo.Application.Apps.Commands.CreateAppCommandHandler.Handle(CreateAppCommand request, CancellationToken cancellationToken) in /Users/frankyang/Project/hippo/src/Application/Apps/Commands/CreateAppCommand.cs:line 39
         at Hippo.Application.Common.Behaviours.PerformanceBehaviour`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next) in /Users/frankyang/Project/hippo/src/Application/Common/Behaviours/PerformanceBehaviour.cs:line 31
         at Hippo.Application.Common.Behaviours.ValidationBehaviour`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next) in /Users/frankyang/Project/hippo/src/Application/Common/Behaviours/ValidationBehaviour.cs:line 35
         at Hippo.Application.Common.Behaviours.AuthorizationBehaviour`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next) in /Users/frankyang/Project/hippo/src/Application/Common/Behaviours/AuthorizationBehaviour.cs:line 78
         at Hippo.Application.Common.Behaviours.UnhandledExceptionBehaviour`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next) in /Users/frankyang/Project/hippo/src/Application/Common/Behaviours/UnhandledExceptionBehaviour.cs:line 19
```

**Comment by bacongobbler** [comment]
Hi @FrankYang0529! Have you re-tried to confirm whether this has been fixed on `main`? I think we shipped a fix for this issue. See https://github.com/deislabs/hippo/pull/876.

**Comment by FrankYang0529** [comment]
Hi @bacongobbler, I tried to test Hippo on `1df9ac` commit, but still get an error. I used spin on `main` branch with https://github.com/fermyon/spin/pull/611.

Spin result:
```
Error: Unable to create Hippo app

Caused by:
    {"type":"https://tools.ietf.org/html/rfc7231#section-6.6.1","title":"An error occurred while processing your request.","status":500,"detail":"TomlException: (4,2) : error : Unable to set the property config on object type Hippo.Application.Common.Interfaces.StorageService.RevisionSpinToml.\n"}
```

Hippo logs:
```
fail: Hippo.Application.Apps.Commands.CreateAppCommand[0]
      Hippo Request: Unhandled Exception for Request CreateAppCommand Hippo.Application.Apps.Commands.CreateAppCommand
      Tomlyn.TomlException: (4,2) : error : Unable to set the property config on object type Hippo.Application.Common.Interfaces.StorageService.RevisionSpinToml.

         at Tomlyn.Toml.ToModel[T](DocumentSyntax syntax, TomlModelOptions options)
         at Hippo.Infrastructure.Services.BindleService.ParseSpinTomlParcel(HttpContent parcelResponse) in /Users/frankyang/Project/hippo/src/Infrastructure/Services/BindleService.cs:line 67
         at Hippo.Infrastructure.Services.BindleService.GetSpinTomlParcel(Invoice invoice, String revisionId) in /Users/frankyang/Project/hippo/src/Infrastructure/Services/BindleService.cs:line 50
         at Hippo.Infrastructure.Services.BindleService.GetRevisionDetails(String revisionId) in /Users/frankyang/Project/hippo/src/Infrastructure/Services/BindleService.cs:line 27
         at Hippo.Application.Revisions.EventHandlers.UpdateRevisionDetails.Handle(CreatedEvent`1 notification, CancellationToken cancellationToken) in /Users/frankyang/Project/hippo/src/Application/Revisions/EventHandlers/UpdateRevisionDetails.cs:line 33
         at MediatR.Mediator.PublishCore(IEnumerable`1 allHandlers, INotification notification, CancellationToken cancellationToken)
         at MediatR.MediatorExtensions.DispatchDomainEvents(IMediator mediator, DbContext context) in /Users/frankyang/Project/hippo/src/Infrastructure/Common/MediatorExtensions.cs:line 22
         at Hippo.Infrastructure.Data.ApplicationDbContext.SaveChangesAsync(CancellationToken cancellationToken) in /Users/frankyang/Project/hippo/src/Infrastructure/Data/ApplicationDbContext.cs:line 62
         at Hippo.Application.Revisions.Commands.ImportRevisionsCommandHandler.Handle(ImportRevisionsCommand request, CancellationToken cancellationToken) in /Users/frankyang/Project/hippo/src/Application/Revisions/Commands/ImportRevisionsCommand.cs:line 49
         at Hippo.Application.Common.Behaviours.PerformanceBehaviour`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next) in /Users/frankyang/Project/hippo/src/Application/Common/Behaviours/PerformanceBehaviour.cs:line 31
         at Hippo.Application.Common.Behaviours.ValidationBehaviour`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next) in /Users/frankyang/Project/hippo/src/Application/Common/Behaviours/ValidationBehaviour.cs:line 35
         at Hippo.Application.Common.Behaviours.AuthorizationBehaviour`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next) in /Users/frankyang/Project/hippo/src/Application/Common/Behaviours/AuthorizationBehaviour.cs:line 78
         at Hippo.Application.Common.Behaviours.UnhandledExceptionBehaviour`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next) in /Users/frankyang/Project/hippo/src/Application/Common/Behaviours/UnhandledExceptionBehaviour.cs:line 19
         at MediatR.Pipeline.RequestExceptionProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
         at MediatR.Pipeline.RequestExceptionProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
         at MediatR.Pipeline.RequestExceptionActionProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
         at MediatR.Pipeline.RequestExceptionActionProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
         at MediatR.Pipeline.RequestPostProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
         at MediatR.Pipeline.RequestPreProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
         at Hippo.Application.Apps.EventHandlers.InitialRevisionImport.Handle(CreatedEvent`1 notification, CancellationToken cancellationToken) in /Users/frankyang/Project/hippo/src/Application/Apps/EventHandlers/InitialRevisionImport.cs:line 26
         at MediatR.Mediator.PublishCore(IEnumerable`1 allHandlers, INotification notification, CancellationToken cancellationToken)
         at MediatR.MediatorExtensions.DispatchDomainEvents(IMediator mediator, DbContext context) in /Users/frankyang/Project/hippo/src/Infrastructure/Common/MediatorExtensions.cs:line 22
         at Hippo.Infrastructure.Data.ApplicationDbContext.SaveChangesAsync(CancellationToken cancellationToken) in /Users/frankyang/Project/hippo/src/Infrastructure/Data/ApplicationDbContext.cs:line 62
         at Hippo.Application.Apps.Commands.CreateAppCommandHandler.Handle(CreateAppCommand request, CancellationToken cancellationToken) in /Users/frankyang/Project/hippo/src/Application/Apps/Commands/CreateAppCommand.cs:line 39
         at Hippo.Application.Common.Behaviours.PerformanceBehaviour`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next) in /Users/frankyang/Project/hippo/src/Application/Common/Behaviours/PerformanceBehaviour.cs:line 31
         at Hippo.Application.Common.Behaviours.ValidationBehaviour`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next) in /Users/frankyang/Project/hippo/src/Application/Common/Behaviours/ValidationBehaviour.cs:line 35
         at Hippo.Application.Common.Behaviours.AuthorizationBehaviour`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next) in /Users/frankyang/Project/hippo/src/Application/Common/Behaviours/AuthorizationBehaviour.cs:line 78
         at Hippo.Application.Common.Behaviours.UnhandledExceptionBehaviour`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next) in /Users/frankyang/Project/hippo/src/Application/Common/Behaviours/UnhandledExceptionBehaviour.cs:line 19
```

**Comment by bacongobbler** [comment]
Thanks for the confirmation. This appears because Hippo does not support the config object in spin.toml. This has been fixed in #1008 and will be made available in the next release. Thanks!
